graph/toposort.cxx

```diff
@@ -0,0 +1,73 @@
+#include <iostream>
+#include <set>
+#include <map>
+#include <queue>
+#include <vector>
+
+template <typename T>
+class Graph {
+	std::map<T, std::set<T>> g;
+public:
+	void addEdge(T a, T b) {
+		g[a].insert(b);
+	}
+	void removeEdge(T a, T b) {
+		g[a].insert(b);
+	}
+
+	std::vector<T> toposort() {
+		std::set<T> visited;
+		std::vector<T> result;
+
+		for(auto const& v : g) {
+			if(visited.find(v.first) == visited.end())
+				dfs(visited, v.first, result);
+		}
+		return result;
+	}
+	void dfs(std::set<T>& visited, const T& v, std::vector<T>& result) {
```

Comment:
This is specifically a post-order dfs and should be named accordingly.  It should also probably be private, it has no use to anyone else.